src/power_grid_model/utils.py

```diff
@@ -88,10 +88,10 @@ def convert_list_to_batch_data(list_data: BatchList) -> BatchDataset:
                 # If the current dataset does not contain the component, add the last indptr again.
                 indptr.append(indptr[-1])
 
-            # Convert the index pointers to a numpy array and combine the list of object numpy arrays into a singe
-            # numpy array. All objects of all batches are now stores in one large array, the index pointers define
-            # which elemets of the array (rows) belong to which batch.
-            batch_data[component] = {"indptr": np.array(indptr, dtype=np.int64), "data": np.concatenate(data, axis=0)}
+        # Convert the index pointers to a numpy array and combine the list of object numpy arrays into a singe
+        # numpy array. All objects of all batches are now stores in one large array, the index pointers define
+        # which elemets of the array (rows) belong to which batch.
+        batch_data[component] = {"indptr": np.array(indptr, dtype=np.int64), "data": np.concatenate(data, axis=0)}
```

Comment:
This is the python utils fix

cause: `np.concatenate` fails when `data` is empty, so the call would fail if the first entry is empty

open question TBD: what to do when all data entries are empty. current implementation will fail.

Possible alternative implementation that **does** allow all entries to be empty: ignore component

```suggestion
        if data:
            batch_data[component] = {"indptr": np.array(indptr, dtype=np.int64), "data": np.concatenate(data, axis=0)}
```

Reply:
I think ignoring component would be nice. Whats your preference?

Reply:
whoops sorry, already clicked merge. let's discuss this during the next knowledge session. the change is very small anyway